streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilder.java

```diff
@@ -1048,24 +1047,23 @@ private void buildProcessorNode(final Map<String, ProcessorNode> processorMap,
     }
 
     private void setRegexMatchedTopicsToSourceNodes() {
-        if (subscriptionUpdates.hasUpdates()) {
-            for (final Map.Entry<String, Pattern> stringPatternEntry : nodeToSourcePatterns.entrySet()) {
-                final SourceNodeFactory sourceNode =
-                    (SourceNodeFactory) nodeFactories.get(stringPatternEntry.getKey());
-                //need to update nodeToSourceTopics with topics matched from given regex
-                nodeToSourceTopics.put(
-                    stringPatternEntry.getKey(),
-                    sourceNode.getTopics(subscriptionUpdates.getUpdates()));
-                log.debug("nodeToSourceTopics {}", nodeToSourceTopics);
+        if (hasSubscriptionUpdates()) {
+            for (final String nodeName : nodeToSourcePatterns.keySet()) {
+                //need to update nodeToSourceTopics and sourceTopicNames with topics matched from given regex
+                final List<String> sourceTopics = ((SourceNodeFactory) nodeFactories.get(nodeName))
+                                                      .getTopics(subscriptionUpdates);
+                nodeToSourceTopics.put(nodeName, sourceTopics);
+                sourceTopicNames.addAll(sourceTopics);
```

Comment:
Also update `sourceTopicNames` to keep in sync with `nodeToSourceTopics`

Reply:
Do we need to do this at this point?  I guess so at it makes sense to have `sourceTopicNames` match what's in `nodeToSourceTopics`.  I'm only asking as we never had this before and I'm curious as to why.

Reply:
Well I did switch `#sourceTopicPattern` and the new `#sourceTopicCollection` to use `sourceTopicNames` instead of `nodeToSourceTopics.values`, in which case we do still need to do this.
It shouldn't matter, although I will say the hardest part of working with this code/class was figuring out which data structures did/did not contain what contents or updates. I'm inclined to leave this in so that the two similar data structures are kept in sync, but I'm fine with removing it just to keep the changes minimal/necessary